cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -318,35 +346,57 @@ private void processMetrics() {
           topicMessageIds.put(topicIdMetaKey, currentMessageId);
         }
 
-        // Skip persisting if the records is empty
-        if (records.isEmpty()) {
-          return;
+        // Try to persist metrics and messageId's of the last metrics to be persisted if no other thread is persisting
+        tryPersist();
+
+        long endTime = System.currentTimeMillis();
+        if (endTime - lastMetricTimeMillis > maxDelayMillis) {
```

Comment:
So if nothing was fetched, the `lastMetricTimeMillis` would still be some older timestamp, hence returning `0L` and causing a non-sleep polling. Is that the intention??